src/App.tsx

```diff
@@ -1,26 +1,302 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  deleteTodos,
+  getTodos,
+  postTodos,
+  updateTodos,
+  updateTodosTitle,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './TodoList';
+import classNames from 'classnames';
+import { Filter } from './Filter';
 
 export const App: React.FC = () => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filtered, setFiltered] = useState('all');
+  const [titleTodo, setTitleTodo] = useState('');
+  const [inputTodo, setInputTodo] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputFocus = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      })
+      .finally(() => {
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (inputFocus.current && inputTodo) {
+      inputFocus.current.focus();
+    }
+  }, [inputTodo]);
+
+  const filteredTodos = todos.filter(todo => {
+    if (filtered === 'active') {
+      return !todo.completed;
+    }
+
+    if (filtered === 'completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  const todosCounter = todos.filter(todo => !todo.completed).length;
+
+  function deletePost(todoId: number): Promise<void> {
+    setInputTodo(false);
+    setIsLoading(true);
+
+    return deleteTodos(todoId)
+      .then(() =>
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        ),
+      )
+      .catch(err => {
+        setErrorMessage('Unable to delete a todo');
+
+        throw err;
+      })
+      .finally(() => {
+        setInputTodo(true);
+        setIsLoading(false);
+      });
+  }
+
+  function addTodo({ userId, title, completed }: Todo) {
```

Comment:
It is important to follow the same way of creating functions, fix it everywhere
```suggestion
  const addTodo = ({ userId, title, completed }: Todo) => {
```